### PR TITLE
Support Swift 5.7 dependencies

### PR DESCRIPTION
**Pull request by giginet**
## Context

Since Xcode 14.0 beta 5, swift-tools-support-core@release/5.6 doesn't build due to broken compatibilities.

I updated `swift-tools-support-core` and `swift-package-manager` to `release/5.7`.
This PR makes this project enable to build with the latest Xcode. (above Xcode 14 beta 5).

## Description

Due to updated dependencies, some APIs are changed. So I changed some implementations.

Unfortunately, GitHub Action doesn't support the latest Xcode yet. So we can't update CI stacks.

### CODE REVIEW

File: Sources/CreateXCFramework/Zipper.swift [hunk 1]
```diff
@@ -106,6 +109,28 @@ struct Zipper {
     func clean (file: Foundation.URL) throws {
         try FileManager.default.removeItem(at: file)
     }
+
+    #if swift(>=5.7)
+    private func checksum(forBinaryArtifactAt path: AbsolutePath) throws -> String {
```
Comment (giginet) [inline Sources/CreateXCFramework/Zipper.swift:114]: `checksum(forBinaryArtifactAt:)` is moved from `Workspace` to `Workspace.BinaryArtifactsManager`.
Unfortunately, `workspace.binaryArtifactsManager` is concealed from externals.

So I ported the same implementation from the original SwiftPM codes.

https://github.com/apple/swift-package-manager/blob/release/5.7/Sources/Workspace/Workspace%2BBinaryArtifacts.swift#L390-L413

File: Sources/CreateXCFramework/ProjectGenerator.swift [hunk 2]
```diff
@@ -23,7 +23,7 @@ struct ProjectGenerator {
 
     var projectPath: AbsolutePath {
         let dir = AbsolutePath(self.package.projectBuildDirectory.path)
-        return buildXcodeprojPath(outputDir: dir, projectName: self.package.manifest.displayName)
+        return XcodeProject.makePath(outputDir: dir, projectName: self.package.manifest.displayName)
```
Comment (giginet) [inline Sources/CreateXCFramework/ProjectGenerator.swift:26]: This global method is moved to a static method.

**Comment by bok-** [comment]
Thanks very much for raising this PR! The code itself looks fantastic to me, I'll run some tests here on my end as well but assuming CI passes we should be good to merge this pretty much straight away.

**Comment by bok-** [comment]
Thanks very much for the PR! I appreciate the time you took to annotate your PR as well – the notes and reasoning made it a really easy one to review.

**Pull request merged**